core/src/display_object/stage.rs

```diff
@@ -0,0 +1,293 @@
+//! Root stage impl
+
+use crate::backend::ui::UiBackend;
+use crate::collect::CollectWrapper;
+use crate::config::Letterbox;
+use crate::context::{RenderContext, UpdateContext};
+use crate::display_object::container::{
+    ChildContainer, DisplayObjectContainer, TDisplayObjectContainer,
+};
+use crate::display_object::{render_base, DisplayObject, DisplayObjectBase, TDisplayObject};
+use crate::prelude::*;
+use crate::types::{Degrees, Percent};
+use gc_arena::{Collect, GcCell, MutationContext};
+
+/// The Stage is the root of the display object hierarchy. It contains all AVM1
+/// levels as well as AVM2 movies.
+#[derive(Clone, Debug, Collect, Copy)]
+#[collect(no_drop)]
+pub struct Stage<'gc>(GcCell<'gc, StageData<'gc>>);
+
+#[derive(Clone, Debug, Collect)]
+#[collect(no_drop)]
+pub struct StageData<'gc> {
+    /// Base properties for all display objects.
+    ///
+    /// This particular base has additional constraints currently not
+    /// expressable by the type system. Notably, this should never have a
+    /// parent, as the stage does not respect it.
+    base: DisplayObjectBase<'gc>,
+
+    /// The list of all children of the stage.
+    ///
+    /// Stage children are exposed to AVM1 as `_level*n*` on all stage objects.
+    child: ChildContainer<'gc>,
+
+    /// The stage background.
+    ///
+    /// If the background color is not specified, it should be white.
+    background_color: CollectWrapper<Option<Color>>,
```

Comment:
Thanks to @Aaron1011, you can add a `#[collect]` attribute directly on the fields now to avoid the external type issue:
```
    #[collect(require_static)]
    background_color: Option<Color>,
```
This can be used any time the type itself is `'static` (`require_static`), which should handle nearly all of our cases. (Lots of older code needs to have this cleaned up still).